bok_choy/a11y/a11y_audit.py

```diff
@@ -0,0 +1,267 @@
+"""
+Interface for running accessibility audits on a PageObject.
+"""
+import json
+import os
+import requests
+
+from abc import abstractmethod, abstractproperty, ABCMeta
+from textwrap import dedent
+
+
+class AccessibilityError(Exception):
+    """
+    The page violates one or more accessibility rules.
+    """
+    pass
+
+
+class A11yAuditConfig(object):
+    """
+    The `A11yAuditConfig` object defines the options available in an
+    accessibility ruleset.
+    """
+    __metaclass__ = ABCMeta
+
+    def __init__(self, *args, **kwargs):
+        super(A11yAuditConfig, self).__init__(*args, **kwargs)
+        self.rules_file = None
+
+    def set_rules_file(self, path=None):
+        """
+        Sets `self.rules_file` to the passed file.
+
+        Args:
+            A filepath where the JavaScript for the ruleset can be found.
+
+        This is intended to be used in the case of using an extended or
+        modified version of the ruleset. The interface and response
+        format are expected to be unmodified.
+        """
+        if path:
+            self.rules_file = os.path.abspath(path)
+
+    @abstractmethod
+    def set_rules(self, rules, ignore=False, include_type="rule"):
```

Comment:
The ignore argument being a boolean flag whose opposite means run was really tripping me up. 
I think the code would be more readable if instead there was a parameter named something like "action" whose argument values could either be "run" or "ignore".